Panache repository persist  failure with  multiple persistence units  in 1.8.0.cr1

**Describe the bug**

  Panache repository persist  failure with  multiple persistence units。
```Java
    @POST
    @Transactional
    public User add(User user) {
        user.setId(null);
        userRepository.persist(user);
        return user;
    }
```
```log
 HTTP Request to /user failed, error id: 97f94515-7d2b-483e-a5d7-940b0ec774af-1: org.jboss.res
teasy.spi.UnhandledException: java.lang.IllegalArgumentException: Not an entity [class com.example.user.User]
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:216)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:515)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:259)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:160)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:163)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:245)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:131)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:37)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:94)
        at io.quarkus.runtime.CleanableExecutor$CleaningRunnable.run(CleanableExecutor.java:231)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.lang.IllegalArgumentException: Not an entity [class com.example.user.User]
        at org.hibernate.internal.SessionImpl.contains(SessionImpl.java:2010)
        at io.quarkus.hibernate.orm.runtime.entitymanager.TransactionScopedEntityManager.contains(TransactionScopedEntityManager.java:286)
        at io.quarkus.hibernate.orm.runtime.entitymanager.ForwardingEntityManager.contains(ForwardingEntityManager.java:122)
        at io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations.persist(AbstractJpaOperations.java:55)
        at io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations.persist(AbstractJpaOperations.java:51)
        at io.quarkus.hibernate.orm.panache.runtime.JpaOperations.persist(JpaOperations.java:58)
        at io.quarkus.hibernate.orm.panache.PanacheRepositoryBase.persist(PanacheRepositoryBase.java:61)
        at com.example.user.UserRepository_ClientProxy.persist(UserRepository_ClientProxy.zig:1196)
        at com.example.user.UserResource.add(UserResource.java:43)
        at com.example.user.UserResource_Subclass.add$$superaccessor1(UserResource_Subclass.zig:211)
        at com.example.user.UserResource_Subclass$$function$$1.apply(UserResource_Subclass$$function$$1.zig:33)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:127)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:100)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:32)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:53)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:26)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:339)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at com.example.user.UserResource_Subclass.add(UserResource_Subclass.zig:168)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:167)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:638)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:504)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:454)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:456)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:417)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:391)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:488)
        ... 20 more
Caused by: org.hibernate.MappingException: Unknown entity: com.example.user.User
        at org.hibernate.metamodel.internal.MetamodelImpl.entityPersister(MetamodelImpl.java:704)
        at org.hibernate.internal.SessionImpl.contains(SessionImpl.java:2007)
        ... 56 more
```

this is a reproducer:
[demo.zip](https://github.com/quarkusio/quarkus/files/5166973/demo.zip)

--- a/extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java
+++ b/extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java
@@ -34,6 +34,7 @@ import io.quarkus.deployment.builditem.ApplicationIndexBuildItem;
 import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.util.JandexUtil;
 import io.quarkus.hibernate.orm.deployment.AdditionalJpaModelBuildItem;
 import io.quarkus.hibernate.orm.deployment.HibernateEnhancersRegisteredBuildItem;
 import io.quarkus.hibernate.orm.deployment.JpaModelPersistenceUnitMappingBuildItem;
@@ -147,18 +148,28 @@ public final class KotlinPanacheResourceProcessor {
 
         KotlinPanacheRepositoryEnhancer daoEnhancer = new KotlinPanacheRepositoryEnhancer(index.getIndex());
         Set<String> daoClasses = new HashSet<>();
+        Set<String> panacheEntities = new HashSet<>();
 
         for (ClassInfo classInfo : index.getIndex().getAllKnownImplementors(PANACHE_REPOSITORY_BASE)) {
             // Skip PanacheRepository
-            if (classInfo.name().equals(PANACHE_REPOSITORY))
+            if (classInfo.name().equals(PANACHE_REPOSITORY)) {
                 continue;
-            if (daoEnhancer.skipRepository(classInfo))
+            }
+            if (daoEnhancer.skipRepository(classInfo)) {
                 continue;
+            }
+            List<org.jboss.jandex.Type> typeParameters = JandexUtil
+                    .resolveTypeParameters(classInfo.name(), PANACHE_REPOSITORY_BASE, index.getIndex());
+            panacheEntities.add(typeParameters.get(0).name().toString());
             daoClasses.add(classInfo.name().toString());
         }
         for (ClassInfo classInfo : index.getIndex().getAllKnownImplementors(PANACHE_REPOSITORY)) {
-            if (daoEnhancer.skipRepository(classInfo))
+            if (daoEnhancer.skipRepository(classInfo)) {
                 continue;
+            }
+            List<org.jboss.jandex.Type> typeParameters = JandexUtil
+                    .resolveTypeParameters(classInfo.name(), PANACHE_REPOSITORY, index.getIndex());
+            panacheEntities.add(typeParameters.get(0).name().toString());
             daoClasses.add(classInfo.name().toString());
         }
         for (String daoClass : daoClasses) {
@@ -204,6 +215,7 @@ public final class KotlinPanacheResourceProcessor {
         }
 
         Map<String, String> panacheEntityToPersistenceUnit = new HashMap<>();
+        panacheEntities.addAll(modelClasses);
 
         if (jpaModelPersistenceUnitMapping.isPresent()) {
             Map<String, Set<String>> collectedEntityToPersistenceUnits = jpaModelPersistenceUnitMapping.get()
@@ -213,7 +225,7 @@ public final class KotlinPanacheResourceProcessor {
             for (Map.Entry<String, Set<String>> entry : collectedEntityToPersistenceUnits.entrySet()) {
                 String entityName = entry.getKey();
                 Set<String> selectedPersistenceUnits = entry.getValue();
-                boolean isPanacheEntity = modelClasses.stream().anyMatch(name -> name.equals(entityName));
+                boolean isPanacheEntity = panacheEntities.stream().anyMatch(name -> name.equals(entityName));
 
                 if (!isPanacheEntity) {
                     continue;

--- a/extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java
@@ -110,17 +110,22 @@ public final class PanacheHibernateResourceProcessor {
     }
 
     @BuildStep
-    void build(CombinedIndexBuildItem index,
+    @Record(ExecutionTime.STATIC_INIT)
+    void build(
+            PanacheHibernateOrmRecorder recorder,
+            CombinedIndexBuildItem index,
             ApplicationArchivesBuildItem applicationArchivesBuildItem,
             BuildProducer<BytecodeTransformerBuildItem> transformers,
             HibernateEnhancersRegisteredBuildItem hibernateMarker,
             List<PanacheEntityClassBuildItem> entityClasses,
-            List<PanacheMethodCustomizerBuildItem> methodCustomizersBuildItems) throws Exception {
+            Optional<JpaModelPersistenceUnitMappingBuildItem> jpaModelPersistenceUnitMapping,
+            List<PanacheMethodCustomizerBuildItem> methodCustomizersBuildItems) {
 
         List<PanacheMethodCustomizer> methodCustomizers = methodCustomizersBuildItems.stream()
                 .map(PanacheMethodCustomizerBuildItem::getMethodCustomizer).collect(Collectors.toList());
 
         PanacheJpaRepositoryEnhancer daoEnhancer = new PanacheJpaRepositoryEnhancer(index.getIndex());
+        Set<String> panacheEntities = new HashSet<>();
         Set<String> daoClasses = new HashSet<>();
         for (ClassInfo classInfo : index.getIndex().getAllKnownImplementors(DOTNAME_PANACHE_REPOSITORY_BASE)) {
             // Skip PanacheRepository
@@ -128,11 +133,17 @@ public final class PanacheHibernateResourceProcessor {
                 continue;
             if (daoEnhancer.skipRepository(classInfo))
                 continue;
+            List<org.jboss.jandex.Type> typeParameters = JandexUtil
+                    .resolveTypeParameters(classInfo.name(), DOTNAME_PANACHE_REPOSITORY_BASE, index.getIndex());
+            panacheEntities.add(typeParameters.get(0).name().toString());
             daoClasses.add(classInfo.name().toString());
         }
         for (ClassInfo classInfo : index.getIndex().getAllKnownImplementors(DOTNAME_PANACHE_REPOSITORY)) {
             if (daoEnhancer.skipRepository(classInfo))
                 continue;
+            List<org.jboss.jandex.Type> typeParameters = JandexUtil
+                    .resolveTypeParameters(classInfo.name(), DOTNAME_PANACHE_REPOSITORY, index.getIndex());
+            panacheEntities.add(typeParameters.get(0).name().toString());
             daoClasses.add(classInfo.name().toString());
         }
         for (String daoClass : daoClasses) {
@@ -186,6 +197,10 @@ public final class PanacheHibernateResourceProcessor {
                 }
             }
         }
+
+        panacheEntities.addAll(modelClasses);
+
+        recordPanacheEntityPersistenceUnits(recorder, jpaModelPersistenceUnitMapping, panacheEntities);
     }
 
     @BuildStep
@@ -205,11 +220,9 @@ public final class PanacheHibernateResourceProcessor {
         return null;
     }
 
-    @BuildStep
-    @Record(ExecutionTime.STATIC_INIT)
-    void persistenceUnits(PanacheHibernateOrmRecorder recorder,
+    void recordPanacheEntityPersistenceUnits(PanacheHibernateOrmRecorder recorder,
             Optional<JpaModelPersistenceUnitMappingBuildItem> jpaModelPersistenceUnitMapping,
-            List<PanacheEntityClassBuildItem> panacheEntityClasses) {
+            Set<String> panacheEntityClasses) {
         Map<String, String> panacheEntityToPersistenceUnit = new HashMap<>();
         if (jpaModelPersistenceUnitMapping.isPresent()) {
             Map<String, Set<String>> collectedEntityToPersistenceUnits = jpaModelPersistenceUnitMapping.get()
@@ -221,7 +234,7 @@ public final class PanacheHibernateResourceProcessor {
                 String entityName = entry.getKey();
                 Set<String> selectedPersistenceUnits = entry.getValue();
                 boolean isPanacheEntity = panacheEntityClasses.stream()
-                        .anyMatch(entity -> entity.get().name().toString().equals(entityName));
+                        .anyMatch(entity -> entity.equals(entityName));
 
                 if (!isPanacheEntity) {
                     continue;

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Entity.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Entity.java
@@ -1,0 +1,37 @@
+package io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.repository;
+
+import java.io.Serializable;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Issue11842Entity implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(length = 64)
+    private String name;
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+}

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Repository.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Repository.java
@@ -1,0 +1,9 @@
+package io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.repository;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+
+@ApplicationScoped
+public class Issue11842Repository implements PanacheRepositoryBase<Issue11842Entity, Integer> {
+}

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Resource.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Resource.java
@@ -1,0 +1,24 @@
+package io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.repository;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+@Path("/repository")
+public class Issue11842Resource {
+    @Inject
+    Issue11842Repository repository;
+
+    @GET
+    @Path("/{name}")
+    @Transactional
+    public String addAndReturnName(@PathParam String name) {
+        Issue11842Entity entity = new Issue11842Entity();
+        entity.setName(name);
+        repository.persist(entity);
+        return repository.findById(entity.getId()).getName();
+    }
+}

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Resource.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/Issue11842Resource.java
@@ -4,21 +4,36 @@ import javax.inject.Inject;
 import javax.transaction.Transactional;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
 
 import org.jboss.resteasy.annotations.jaxrs.PathParam;
 
-@Path("/repository")
+import io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.first.FirstEntity;
+
+@Path("/persistence-unit")
 public class Issue11842Resource {
     @Inject
     Issue11842Repository repository;
 
     @GET
-    @Path("/{name}")
+    @Path("/repository/{name}")
     @Transactional
-    public String addAndReturnName(@PathParam String name) {
+    public Long addAndReturnCountUsingPanacheRepository(@PathParam String name) {
         Issue11842Entity entity = new Issue11842Entity();
         entity.setName(name);
         repository.persist(entity);
-        return repository.findById(entity.getId()).getName();
+        return repository.count();
+    }
+
+    @GET
+    @Path("/panache-entity/{name}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public Long addAndReturnCountUsingPanacheEntity(@PathParam String name) {
+        FirstEntity entity = new FirstEntity();
+        entity.name = name;
+        entity.persistAndFlush();
+        return FirstEntity.count();
     }
 }

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/MultiplePersistenceUnitConfigForRepositoryTest.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/MultiplePersistenceUnitConfigForRepositoryTest.java
@@ -1,0 +1,27 @@
+package io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.repository;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.first.FirstEntity;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class MultiplePersistenceUnitConfigForRepositoryTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(Issue11842Entity.class, Issue11842Repository.class, Issue11842Resource.class, FirstEntity.class)
+                    .addAsResource("application-multiple-persistence-units-for-repository.properties",
+                            "application.properties"));
+
+    @Test
+    public void panacheOperations() {
+        RestAssured.when().get("/repository/name-1").then().body(Matchers.is("name-1"));
+        RestAssured.when().get("/repository/name-2").then().body(Matchers.is("name-2"));
+    }
+}

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/MultiplePersistenceUnitConfigForRepositoryTest.java
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/java/io/quarkus/hibernate/orm/panache/deployment/test/multiple_pu/repository/MultiplePersistenceUnitConfigForRepositoryTest.java
@@ -21,7 +21,12 @@ public class MultiplePersistenceUnitConfigForRepositoryTest {
 
     @Test
     public void panacheOperations() {
-        RestAssured.when().get("/repository/name-1").then().body(Matchers.is("name-1"));
-        RestAssured.when().get("/repository/name-2").then().body(Matchers.is("name-2"));
+        // Using PanacheRepository
+        RestAssured.when().get("/persistence-unit/repository/name-1").then().body(Matchers.is("1"));
+        RestAssured.when().get("/persistence-unit/repository/name-2").then().body(Matchers.is("2"));
+
+        // Using PanacheEntity
+        RestAssured.when().get("/persistence-unit/panache-entity/name-1").then().body(Matchers.is("1"));
+        RestAssured.when().get("/persistence-unit/panache-entity/name-2").then().body(Matchers.is("2"));
     }
 }

--- a/extensions/panache/hibernate-orm-panache/deployment/src/test/resources/application-multiple-persistence-units-for-repository.properties
+++ b/extensions/panache/hibernate-orm-panache/deployment/src/test/resources/application-multiple-persistence-units-for-repository.properties
@@ -1,0 +1,14 @@
+quarkus.datasource.db-kind=h2
+quarkus.datasource.jdbc.url=jdbc:h2:mem:default;DB_CLOSE_DELAY=-1
+
+quarkus.datasource.repository.db-kind=h2
+quarkus.datasource.repository.jdbc.url=jdbc:h2:mem:repository;DB_CLOSE_DELAY=-1
+
+quarkus.hibernate-orm.dialect=org.hibernate.dialect.H2Dialect
+quarkus.hibernate-orm.database.generation=drop-and-create
+quarkus.hibernate-orm.packages=io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.first
+
+quarkus.hibernate-orm."repository".dialect=org.hibernate.dialect.H2Dialect
+quarkus.hibernate-orm."repository".database.generation=drop-and-create
+quarkus.hibernate-orm."repository".datasource=repository
+quarkus.hibernate-orm."repository".packages=io.quarkus.hibernate.orm.panache.deployment.test.multiple_pu.repository

